tests/test_multipart_operations.py

```diff
@@ -0,0 +1,99 @@
+import logging
+
+from common_ci_utils.random_utils import (
+    generate_random_hex,
+    generate_unique_resource_name,
+    generate_random_files,
+)
+from utility.utils import (
+    check_data_integrity,
+    split_file_data_for_multipart_upload,
+)
+
+log = logging.getLogger(__name__)
+
+
+def test_s3_multipart_operations(
+    account_manager,
+    s3_client_factory,
+    tmp_directories_factory,
+):
+    """
+    Test basic s3 operations using a noobaa bucket:
+    1. Create an account
+    2. Create a bucket using S3
+    3. Write multipart objects to the bucket
+    4. List multipart objects from the bucket
+    5. Download the objects from the bucket and verify data integrity
+    6. Delete the bucket along with objects using S3
+    """
+
+    origin_dir, results_dir = tmp_directories_factory(
+        dirs_to_create=["origin", "result"]
+    )
+
+    # 1. Create an account uisng Node CLI
+    account_name = generate_unique_resource_name(prefix="account")
+    access_key = generate_random_hex()
+    secret_key = generate_random_hex()
+    account_manager.create(account_name, access_key, secret_key)
+    s3_client = s3_client_factory(
+        access_and_secret_keys_tuple=(access_key, secret_key)
+    )
+    # 2. Create a bucket using S3
+    bucket_name = s3_client.create_bucket()
+    assert bucket_name in s3_client.list_buckets(), "Bucket was not created"
+    log.info("Bucket created successfully")
+
+    # 3. Write multipart objects to the bucket
+    object_names = generate_random_files(
+        origin_dir,
+        amount=1,
+        min_size="20M",
+        max_size="30M",
+    )
+    # Upload multipart object
+    for i in range(len(object_names)):
```

Comment:
This would be a cleaner way to do this:

```suggestion
    for obj in in object_names:
```

And if you do need the index, you can use:
```suggestion
    for i, obj in enumerate(object_names):
```